src/pages/blog/styling-in-react.md

```diff
@@ -0,0 +1,158 @@
+
+## create-react-app
+
+If you have used [create-react-app](https://github.com/facebook/create-react-app) you’ve seen that it creates a separate file for App.js and for App.css, and for index.js and index.css. Do you think create-react-app promotes separating the concerns in a way that is easier to maintain our code?
+
+<img placeholder-height="px" src="https://firebasestorage.googleapis.com/v0/b/reactjsacademy-react.appspot.com/o/blog%20post%20images%2Fstyling%2Fcreate-react-app-min.png?alt=media" alt="Create React App css  approach"></img>
+
+I think it does. It follows the co-location principle, it co-locates the JS and the CSS close. Note, we also apply the co-location principle in React Router and GraphQL for instance.
+
+So, are we done with styling? Well, there is another issue created by global styles. That folder structure separates the code by concern, but nothing stops us from creating an “active” class in Button.css, and an “active” class in Alert.css with different values. What do you think it’s going to happen if we have an active button and an active alert in the same view?
+
+<img placeholder-height="px" src="https://firebasestorage.googleapis.com/v0/b/reactjsacademy-react.appspot.com/o/blog%20post%20images%2Fstyling%2Fcollision-min.png?alt=media" alt="CSS collision"></img>
+
+<img placeholder-height="px" src="https://firebasestorage.googleapis.com/v0/b/reactjsacademy-react.appspot.com/o/blog%20post%20images%2Fstyling%2Fblue-or-orange-min.png?alt=media" alt="Blue button or orange button?"></img>
+
+## Naming conventions
+
+The solution to the style collision is to use name conventions in our CSS classes. There are different conventions. The problem with name conventions is that humans must implement them. That’s more work, and it’s error-prone.
+
+<img placeholder-height="px" src="https://firebasestorage.googleapis.com/v0/b/reactjsacademy-react.appspot.com/o/blog%20post%20images%2Fstyling%2Fnaming-conventions-min.png?alt=media" alt="CSS naming conventions"></img>
+
+A better approach is to automate that CSS naming. You can use CSS modules for this:
+
+<img placeholder-height="px" src="https://firebasestorage.googleapis.com/v0/b/reactjsacademy-react.appspot.com/o/blog%20post%20images%2Fstyling%2Fcss-modules-min.png?alt=media" alt="CSS modules"></img>
+
+[css-loader](https://github.com/webpack-contrib/css-loader) is the loader that makes CSS modules work. css-loader is installed and configured by default when using create-react-app.
+
+## Styled-components <a name="styled-components"></a>
+
+[Styled-components](https://www.styled-components.com/) is the library we use at LeanJS to style our React components. Why? A single library that solves elegantly all the problems I’ve mentioned.
+
+<img placeholder-height="px" src="https://firebasestorage.googleapis.com/v0/b/reactjsacademy-react.appspot.com/o/blog%20post%20images%2Fstyling%2Fstyled-components-check-list-min.png?alt=media" alt="Styled-components features"></img>
```

Comment:
We really need to replace all the pictures with text in them for _actual text_. There's no need to have the text in an image especially in this case as it's just a list of advantages that - if you can't see the image - makes the blog skip vital content.